src/Infrastructure/DistGrid/src/ESMCI_DistGrid.C

```diff
@@ -4962,6 +4962,52 @@ ESMC_I8 DistGrid::getElementCountPDe(
 }
 //-----------------------------------------------------------------------------
 
+//-----------------------------------------------------------------------------
+#undef  ESMC_METHOD
+#define ESMC_METHOD "ESMCI::DistGrid::getTilePLocalDe()"
+//BOPI
+// !IROUTINE:  ESMCI::DistGrid::getTilePLocalDe
+//
+// !INTERFACE:
+int DistGrid::getTilePLocalDe(
```

Comment:
Note this new function. This allows replacing a multi-step process (converting local DE to global and then using that to index into an array) with a single function call. I didn't replace all possible uses: I focused on uses of the old pattern where we wouldn't otherwise need to do the local to global DE conversion, because that's where the simplification is greatest.